crates/router/src/connector/stripe/transformers.rs

```diff
@@ -287,8 +430,14 @@ impl TryFrom<&types::VerifyRouterData> for SetupIntentRequest {
         let metadata_txn_id = format!("{}_{}_{}", item.merchant_id, item.payment_id, "1");
         let metadata_txn_uuid = Uuid::new_v4().to_string();
 
-        let payment_data: StripePaymentMethodData =
-            (item.request.payment_method_data.clone(), item.auth_type).try_into()?;
+        //Only cards supported for mandates
+        let pm_type = StripePaymentMethodType::Card;
+        let payment_data: StripePaymentMethodData = (
+            item.request.payment_method_data.clone(),
+            item.auth_type,
+            pm_type,
+        )
+            .try_into()?;
```

Comment:
In this case, I feel `try_from()` is more readable. Not a necessary change, you're free to discard this suggestion.

```suggestion
        let payment_data = StripePaymentMethodData::try_from(
            item.request.payment_method_data.clone(),
            item.auth_type,
            pm_type,
        )?;
```

Reply:
I personally feel we should never use try_from from types like tuple, this will be error prone because it requires field ordering. There should either be a function or a type which encapsulates the tuple